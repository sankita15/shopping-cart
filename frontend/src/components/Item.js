import React from 'react';
import { Card, CardImg, CardBody, CardTitle, CardText, CardSubtitle } from 'reactstrap';
import * as PropTypes from 'prop-types';

const Item = ({ imageUrl, description, price, productName, id }) => (
    <a href={`/products/${id}`}>
        <div className="item-page">
            <Card className="item-card">
                <CardImg top width="100%" className="item-page-image" src={imageUrl} />
                <CardBody className="item-body">
                    <CardTitle>{productName}</CardTitle>
                    <CardSubtitle>
                                Rs.
                        {price}
                    </CardSubtitle>
                    <CardText>{description}</CardText>
                </CardBody>
            </Card>
        </div>
    </a>
);

Item.propTypes = {
    id: PropTypes.string.isRequired,
    productName: PropTypes.string.isRequired,
    description: PropTypes.string.isRequired,
    price: PropTypes.number.isRequired,
    imageUrl: PropTypes.string.isRequired,
};

export default Item;
