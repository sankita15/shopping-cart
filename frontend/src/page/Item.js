import React from 'react';
import {Card, CardImg, CardBody, CardTitle, CardText, CardSubtitle, Button, CardDeck} from "reactstrap";
import PropTypes from 'prop-types';

export default class Item extends React.Component {

    render() {
        const {item} = this.props;
        return (
            <div className="item-page">
                    <Card className="item-card">
                        <CardImg top width="100%" className="item-image" src={item.imageUrl}/>
                        <CardBody className="item-body">
                            <CardTitle>{item.productName}</CardTitle>
                            <CardSubtitle>Rs.{item.price}</CardSubtitle>
                            <CardText>{item.description}</CardText>
                            <Button className="btn btn-primary">View Details</Button>
                        </CardBody>
                    </Card>
            </div>
        )
    }
}

Item.propTypes = {
    item: PropTypes.arrayOf(PropTypes.shape({
        id: PropTypes.string,
        productName: PropTypes.string,
        productCode: PropTypes.string,
        releaseDate: PropTypes.number,
        lastModified: PropTypes.number,
        description: PropTypes.string,
        starRating: PropTypes.number,
        price: PropTypes.number,
        imageUrl: PropTypes.string,
        stock: PropTypes.number
    }))
};
