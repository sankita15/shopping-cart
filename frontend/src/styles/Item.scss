.item-page {

  width: 25%;
  height: 341px;
  float: left;

  .item-card {
    margin: 10px;
    box-shadow: 0 4px 8px 0 rgba(0, 0, 0, 0.2);
    transition: 0.3s;

    .item-page-image {
      width: 200px;
      height: 200px;
      margin-left: 70px;
    }

    .item-body{
      color: black;
    }

    .btn-secondary {
      background-color: #007bff;
      border-color: #007bff;
    }

    .btn .btn-primary .btn-secondary {
      width: 100%;
      display: block;
    }

  }
}
