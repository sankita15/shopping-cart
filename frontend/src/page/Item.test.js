import React from 'react';
import { shallow } from 'enzyme';
import { Card, CardImg, CardBody, CardTitle, CardSubtitle, CardText } from 'reactstrap';
import Item from './Item';

describe('Item page', () => {
    const defaultProps = {
        id: '5c61496906057ea38af387cd',
        productName: 'Leaf Rake',
        description: 'Leaf rake with 48-inch wooden handle.',
        price: 1995,
        imageUrl: 'http://openclipart.org/image/300px/svg_to_png/26215/Anonymous_Leaf_Rake.png',
    };

    const createWrapper = () => shallow(<Item {...defaultProps} />);

    it('should find card for each Item', () => {
        const component = createWrapper();
        const card = component.find(Card);
        const cardImg = component.find(Card).find(CardImg);
        const cardBody = component.find(Card).find(CardBody);

        expect(card.length).toBe(1);
        expect(cardImg.length).toBe(1);
        expect(cardBody.length).toBe(1);
        expect(cardBody.find(CardTitle).length).toBe(1);
        expect(cardBody.find(CardSubtitle).length).toBe(1);
        expect(cardBody.find(CardText).length).toBe(1);
    });

    it('should match snapshot', () => {
        const component = createWrapper();

        expect(component).toMatchSnapshot();
    });

    it('should redirect to product details on clicking of product card', () => {
        const component = createWrapper();
        const anchorClick = component.find('a').prop('href');

        expect(anchorClick).toBe(`/products/${defaultProps.id}`);
    });
});
