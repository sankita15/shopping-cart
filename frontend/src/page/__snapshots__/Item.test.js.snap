// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Item page should match snapshot 1`] = `
<a
  href=""
  onClick={[Function]}
>
  <div
    className="item-page"
  >
    <Card
      className="item-card"
      tag="div"
    >
      <CardImg
        className="item-image"
        src="http://openclipart.org/image/300px/svg_to_png/26215/Anonymous_Leaf_Rake.png"
        tag="img"
        top={true}
        width="100%"
      />
      <CardBody
        className="item-body"
        tag="div"
      >
        <CardTitle
          tag="div"
        >
          Leaf Rake
        </CardTitle>
        <CardSubtitle
          tag="div"
        >
          Rs.
          1995
        </CardSubtitle>
        <CardText
          tag="p"
        >
          Leaf rake with 48-inch wooden handle.
        </CardText>
      </CardBody>
    </Card>
  </div>
</a>
`;
